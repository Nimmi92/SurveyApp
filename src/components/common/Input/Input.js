import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { device } from '../../../device/device';

const StyledLabel = styled.label`
	display: inline-block;
	margin: 0 5px 10px 0;

	@media ${device.mobileL} {
      display: block;
      margin: 0 0 10px 0;
      text-align: left;
  	}
`
const StyledInput = styled.input`
	display: inline-block;
	padding: 2px 5px;
	background: #fff;
	margin: 0 0 10px 0;
	background-color: #fafbfc;
    border: 1px solid #d1d5da;
    border-radius: 3px;
    height: 20px;
    box-shadow: inset 0 1px 2px rgba(27,31,35,.075);

	@media ${device.mobileL} {
      display: block;
      width: 95%;
  	}
`
const Input = (props) => {
  const {data} = props;
  return (
    <div>
     { data.label && <StyledLabel htmlFor={data.id}> {data.label} </StyledLabel> }
     <StyledInput type={data.type} placeholder={data.placeholder} id={data.id} size={data.size} />
     
    </div>
  );
};


const { string } = PropTypes;

Input.propTypes = {
	label: string,
	id: string,
	type: string.isRequired,
	placeholder: string,
	size: string
}

export default Input;
