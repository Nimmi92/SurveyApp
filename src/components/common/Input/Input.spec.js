import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';
import toJson from "enzyme-to-json";

describe('<Input />', () => {
  it('should be an input element', () => {
    const props = {
      placeholder: 'Please enter repo name',
      type: 'text',
      size: '30'
    };

    const wrapper = shallow(<Input {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

})