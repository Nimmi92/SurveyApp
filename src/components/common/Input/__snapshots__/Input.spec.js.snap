// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should be an input element with given id 1`] = `
<div>
  <ForwardRef
    htmlFor="repoName"
  >
     
    Repo Name
     
  </ForwardRef>
  <ForwardRef
    id="repoName"
    placeholder="Please enter repo name"
    size="30"
    type="text"
  />
</div>
`;
